internal/graph/graph.go

```diff
@@ -144,6 +145,8 @@ type NodeMap map[NodeInfo]*Node
 // NodeSet maps is a collection of node info structs.
 type NodeSet map[NodeInfo]bool
 
+type nodePtrSet map[*Node]bool
```

Comment:
Let's export this type. It is awkward to have exported functions return unexported types